app/src/main/java/org/cru/godtools/widget/AnalyticsMovementMethod.kt

```diff
@@ -0,0 +1,52 @@
+package org.cru.godtools.widget
+
+import android.net.Uri
+import android.text.Spannable
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.text.style.URLSpan
+import android.view.MotionEvent
+import android.widget.TextView
+import org.cru.godtools.base.tool.analytics.model.ExitLinkActionEvent
+import org.greenrobot.eventbus.EventBus
+import timber.log.Timber
+
+class AnalyticsMovementMethod : LinkMovementMethod() {
+
+    private val eventBus by lazy { EventBus.getDefault() }
+
+    override fun onTouchEvent(widget: TextView?, buffer: Spannable?, event: MotionEvent?): Boolean {
+        Timber.tag("AnalyticsMovement").i("$widget $buffer $event")
+
+        var x = event?.x?.toInt() ?: 0
+        var y = event?.y?.toInt() ?: 0
+
+        x -= widget?.totalPaddingLeft ?: 0
+        y -= widget?.totalPaddingTop ?: 0
+
+        x += widget?.scrollX ?: 0
+        y += widget?.scrollY ?: 0
+        val layout = widget?.layout
+        val line = layout?.getLineForVertical(y) ?: 0
+        val off = layout?.getOffsetForHorizontal(line, x.toFloat()) ?: 0
+
+        val links = buffer?.getSpans(off, off, ClickableSpan::class.java)
+        links?.forEach {
+            if (it is URLSpan) {
+                it.url.let { url ->
+                    eventBus.post(ExitLinkActionEvent(Uri.parse(url)))
+                }
+            }
+        }
+        return super.onTouchEvent(widget, buffer, event)
+    }
```

Comment:
this is a very hacky way to trigger the analytics, I would rather see us use a custom `URLSpan` class if possible.